submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireFilesService.kt

```diff
@@ -9,25 +10,43 @@ import ebi.ac.uk.extended.model.ExtSubmission
 import ebi.ac.uk.extended.model.FireDirectory
 import ebi.ac.uk.extended.model.FireFile
 import ebi.ac.uk.extended.model.NfsFile
+import ebi.ac.uk.extended.model.allFiles
 import ebi.ac.uk.io.ext.md5
 import ebi.ac.uk.io.ext.size
 import mu.KotlinLogging
 import uk.ac.ebi.fire.client.integration.web.FireWebClient
 
 private val logger = KotlinLogging.logger {}
 
-class FireFilesService(private val fireWebClient: FireWebClient) : FilesService {
+class FireFilesService(
+    private val fireWebClient: FireWebClient,
+    private val submissionQueryService: SubmissionQueryService
+) : FilesService {
     override fun persistSubmissionFiles(request: FilePersistenceRequest): ExtSubmission {
         val (sub, _, previousFiles) = request
         logger.info { "${sub.accNo} ${sub.owner} Persisting files of submission ${sub.accNo} on FIRE" }
 
+        cleanFtpFolder(sub)
         val config = FireFileProcessingConfig(sub.accNo, sub.owner, sub.relPath, fireWebClient, previousFiles)
         val processed = processFiles(sub) { config.processFile(request.submission, it) }
 
         logger.info { "${sub.accNo} ${sub.owner} Finished persisting files of submission ${sub.accNo} on FIRE" }
 
         return processed
     }
+
+    private fun cleanFtpFolder(submission: ExtSubmission) {
```

Comment:
cleaning the FTP folder here does not look correct for me. If you notice we have two interfaces FtpService and FilesService, the first one is in charge of ftp operation and the second one is in charge of Ftp operation. As this class is the implementation of the first one, no ftp concer should be included here. 

I think what we need is to change ftp service implementation to unpublish old files.

Reply:
Yes, perhaps the naming was poor, it should be clean submission folder